src/analysis/complex/specific_limits.lean

```diff
@@ -0,0 +1,79 @@
+/-
+Copyright (c) 2022 Dylan MacKenzie. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Dylan MacKenzie
+-/
+
+import analysis.complex.basic
+import analysis.normed.group.infinite_sum
+import analysis.specific_limits
+
+/-!
+# A collection of specific limit computations involving complex numbers
+-/
+
+open filter finset asymptotics
+open_locale nat topological_space big_operators
+
+section is_R_or_C
+
+variables {E : Type*} [is_R_or_C E]
+variables {x : ℝ} {f : ℕ → ℝ} {z : ℕ → E}
+
+/-- **Dirichlet's Test** for monotone sequences. -/
+theorem cauchy_seq_series_mul_of_monotone_tendsto_zero_of_series_bounded
+  (hfa : monotone f) (hf0 : tendsto f at_top (𝓝 0)) (hgx : ∀ n, ∥∑ i in range n, z i∥ ≤ x) :
+  cauchy_seq (λ n, ∑ i in range (n+1), ↑(f i) * z i) :=
+begin
+  simp_rw [sum_range_by_parts _ _ (nat.succ_pos _), sub_eq_add_neg,
+           nat.succ_sub_succ_eq_sub, tsub_zero],
+  apply cauchy_seq.add _ _,
+  { exact normed_uniform_group },
```

Comment:
It feels like typeclass inference should handle these. I saved the instance with `haveI`, but it didn't help.

Reply:
This is due to the way `apply` triggers type class search at the wrong time (before having enough information to success). Replacing `apply` by `refine` would fix it (in your case you don't even really use `apply` since you put trailing underscores).